blogposts/2021/how-we-built-batch-changes.md

```diff
@@ -0,0 +1,202 @@
+---
+title: 'How we built Batch Changes, explained in 5 pull requests'
+description: A 300 character limit field for describing your post. Use this is you want to specially craft the excerpt shown on the index page. Uses the first 300 characters of text from your post if this field does not exist.
+author: Thorsten Ball
+authorUrl: https://twitter.com/thorstenball
+publishDate: 2021-04-16T18:00+02:00
+tags: [blog]
+slug: how-we-built-batch-changes
+# TODO: heroImage: /blog/TODO.jpg
+# TODO: socialImage: https://about.sourcegraph.com/blog/TODO.jpg
+published: false
+---
+
+"Straightforward" is the last word I'd use to describe how we built Batch
+Changes. Exciting, rewarding, surprising, iterative, customer-focused - one of
+those maybe, but not straightfoward. The two name changes in 1.5 years of
+development — from Automation to Campaigns to Batch Changes — give you a glimpse
+of why. The real story, though, the one that makes me doubt whether we actually
+built and shipped Batch Changes the way we did, I think is best told by five
+pull requests.
+
+Not because these pull requests solve hard technical problems in two lines of
+clever code or fix nasty bugs that held us back for months. It's because they
+show how we avoided two big dangers of iteration-based product development:
+local maximums and the sunken cost fallacy.
+
+To give you some context for these terms, let me tell you how I understand the
+premise of iteration-based work. It doesn't matter whether you call it _Agile_,
+_Sprint_, _Shape Up_ or something else, the idea is this: until you've built the
+product, you don't _really_ know what it should look like or how to best build
+it. You've never used it — how would you know how it works best? You also
+haven't built it yet, which means you don't have experience building it. Chances
+are low that you can accurately predict how its development will go.
+
+So what you do is you acknowledge that you don't know. And then you lean into that:
+you _don't_ plan the development of the product from start to finish in every
+detail. You _don't_ nail down and define the hundreds of steps it will take to
+go from zero to fully-built and shipped and launched product. Instead, working
+in iterations, you define roughly where you want to go and then take a single
+small step. Then you look around, get your bearings, correct course if
+necessary, take another step. Build something, ship it, learn, build again.
```

Comment:
> Speaking as the author of the post you don't need to add qualifiers.

I still would like to leave them in, since I don't think it's a universal explanation for the concept (others _have_ different definitions) but a rather subjective view.

Reply:
I moved the paragraphs up, incorporated some wording suggestions, but left others in. Let me know if that works.